src/src/Helpers/_CPlugin_Helper_mqtt.cpp

```diff
@@ -591,7 +738,12 @@ bool MQTT_DiscoveryGetDeviceVType(taskIndex_t                 TaskIndex,
     case 144: // Vindriktning / PM1006K
       discoveryItems.push_back(DiscoveryItem(Sensor_VType::SENSOR_TYPE_DUSTPM2_5_ONLY, valueCount, 0));
       break;
-    case 47:  // Soil Moisture
+    case 27:  // INA219
```

Comment:
Not sure if it is the correct place to do these plugin specifics here.
Maybe best to 'ask' the plugin (or task) to format those. A bit like the raw option as used for LoRaTTN.

This way the knowledge remains inside the plugin code and you don't het those ugly `#ifdef USES_Pxxx` checks for each case in this switch statement.

Reply:
At first I was quite reluctant to change (nearly) all plugins, but it's indeed the ~best~ right way to go forward. (And I practiced on changing all those files, recently 😅)
I've already added a plugin function `PLUGIN_GET_DISCOVERY_VTYPES` to retrieve that info (see P052 for a custom/user-defined implementation, for standard plugins it's much simpler)

I've 'configured' most of the standard plugins in a single place for now, making that process easier. I'll move that to the separate files 😉 👍

Reply:
Maybe this can later also be used to implement the feature to use UnitOfMeasure.
This can then be used to offer default sensible settings for the stats/chart layout, unit conversion (C-to-F for example) and some standard options to globally set preferred units.

So internally we will use standard Si units and when using the value it will be converted to the requested unit of measure.

Reply:
> this can later also be used to implement the feature to use UnitOfMeasure.

That's one of the things that keeps popping up in the back of my head while assembling those Json discovery excerpts 😃